src/js/utils/specs/date-utils-spec.js

```diff
@@ -0,0 +1,145 @@
+                });
+            });
+
+            describe('getSysTimezone', function() {
+                it('attempts to determine users timezone', function() {
+                    // we need to determine the system tz offset from which to compare
+                    var sysDate = new Date();
+                    var sysDateOffset = sysDate.getTimezoneOffset();
+                    var dateUtilOffset = momentTZ.tz.zone(DateUtils.getSysTimezone()).offset(sysDate);
+                    expect(sysDateOffset).toEqual(dateUtilOffset);
+                });
+            });
+
+            describe('localizeTime', function() {
+                it('tests error conditions', function() {
+                    expect(DateUtils.localizeTime()).toEqual('');
+                });
+                it('tests sys time determiner', function() {
+                    var testDate = DateUtils.stringToMoment('2016-10-14 08:00:00+00:00');
+                    expect(
+                         moment(DateUtils.localizeTime(testDate).format()).isValid()
+                         ).toBeTruthy();
+                });
+                it('offsets time according to a determined timezone', function() {
+                    var testDate = DateUtils.stringToMoment('2016-10-14 08:00:00+00:00');
+                    // in order to avoid DST messes, we'll just pick some TZs without
+                    var TestZones = {
+                        'America/Winnipeg': 'Oct 14, 2016 3:00 AM',
+                        'Pacific/Honolulu': 'Oct 13, 2016 10:00 PM',
+                        'Asia/Ho_Chi_Minh': 'Oct 14, 2016 3:00 PM',
+                        'Atlantic/Reykjavik': 'Oct 14, 2016 8:00 AM'
+                    };
+                    Object.keys(TestZones).forEach(function(key) {
+                        expect(
+                            new DateUtils.localizeTime(testDate, key).format('lll')
+                        ).toEqual(TestZones[key]);
+                    });
+                });
+            });
+
+            describe('localize', function() {
+                it('tests error conditions', function() {
+                    var context = {};
+                    expect(
+                        DateUtils.localize(context)
+                    ).toEqual('');
+                });
+                it('tests defaults', function() {
+                    var context = {
+                        datetime: '2016-10-14 08:00:00+00:00',
+                        timezone: 'RANDOM_STRING',
+                        language: 'RANDOM_STRING'
+                    };
+                    expect(DateUtils.localize(context)).toEqual('Oct 14, 2016 08:00 UTC');
+                    context = {
+                        datetime: 'RANDOM_STRING',
+                        timezone: 'RANDOM_STRING',
+                        language: 'RANDOM_STRING'
+                    };
+                    expect(DateUtils.localize(context)).toEqual('');
```

Comment:
Nittiest of the nit: this should be moved to "tests error conditions" instead of "tests defaults", right?